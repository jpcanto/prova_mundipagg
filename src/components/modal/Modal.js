import React from 'react';

const Modal = (props) => {
    return (
        <div className="modal">
            {
                props.state.onAppClose ?
                    <div className="modal__message-box">
                        <p>Você está finalizando o app, todas as alterações serão perdidas!</p>
                    </div>
                    :
                    null
            }
            {
                props.state.isValidate ?
                    <div className="modal__succes-box">
                        <p>Parabéns, suas informações foram salvas com sucesso!</p>
                        <p>Informações do app:</p>
                        <div className="modal__succes-box__app-info">
                            <p>{`Nome: ${props.state.appName}`}</p>
                            <p>{`Categoria: ${props.state.appCategory}`}</p>
                            <p style={{color: props.state.mudaCor}}>{`Cor: ${props.state.mudaCor}`}</p>
                            <p>{`Url da imagem: ${props.state.files[0].preview.url.slice(5, props.state.files[0].preview.url.lastIndexOf('.png' || '.jpg' || '.svg'))}`}</p>
                        </div>
                    </div>
                    :
                    null
            }
        </div>
    );
}

export default Modal;

