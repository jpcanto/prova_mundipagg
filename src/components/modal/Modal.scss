.modal {
    color: #fff;
    font-family: sans-serif;
    font-size: 1.5em;
    
    align-items: center;
    background-color: #000000bd;
    bottom: 0;
    display: flex;
    left: 0;
    justify-content: center;
    position: fixed;
    right: 0;
    top: 0;
    z-index: 2;
    &__message-box,
    &__succes-box {
        background: #4d4d4d;
        padding: 15vh 10vw;
        position: absolute;
        top: 12%;
    }
    &__succes-box {
        color: #fff;

        background: #4d4d4d;
        border-radius: 10px;
        & > p:first-child {
            font-weight: bold;

            border-bottom: 1px solid #5d5d5d57;
        }
        & > p:nth-child(2) {
            text-align: center;
        }
        &__app-info {
            text-align: center;

            border: 1px dashed;
            border-radius: 5px;
            padding: 15px;
        }
    }
}