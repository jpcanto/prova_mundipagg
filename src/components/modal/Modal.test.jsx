import React from 'react';
import Modal from './Modal.js';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';
import toJson from "enzyme-to-json";

configure({adapter: new Adapter()});

describe('Testando renderização do componente Modal', () => {
    it("Deve renderizar corretamente", () => {

        const state = {
            onAppClose: true,
        }

        const wrapper = mount(
            <Modal state={state}/>
        );

        expect(toJson(wrapper)).toMatchSnapshot()

    });
});