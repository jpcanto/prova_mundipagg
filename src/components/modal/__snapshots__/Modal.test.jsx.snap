// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando renderização do componente Modal Deve renderizar corretamente 1`] = `
<Modal
  state={
    Object {
      "onAppClose": true,
    }
  }
>
  <div
    className="modal"
  >
    <div
      className="modal__message-box"
    >
      <p>
        Você está finalizando o app, todas as alterações serão perdidas!
      </p>
    </div>
  </div>
</Modal>
`;
